core/src/main/scala/kafka/admin/ConsumerGroupCommand.scala

```diff
@@ -55,83 +57,137 @@ object ConsumerGroupCommand {
 
     val consumerGroupService = {
       if (opts.useOldConsumer) {
-        new ZkConsumerGroupService(opts)
+        println("Note: This will only show information about consumers that use ZooKeeper (not those using the Java consumer API).\n")
+        new ZkConsumerGroupService(opts, consumerGroupOutputWriter)
       } else {
+        println("Note: This will only show information about consumers that use the Java consumer API (non-ZooKeeper-based consumers).\n")
         new KafkaConsumerGroupService(opts)
       }
     }
 
     try {
       if (opts.options.has(opts.listOpt))
-        consumerGroupService.list()
-      else if (opts.options.has(opts.describeOpt))
-        consumerGroupService.describe()
+        consumerGroupService.listGroups().foreach { consumerGroupOutputWriter.printMessage(_) }
+      else if (opts.options.has(opts.describeOpt)) {
+        consumerGroupService.describeGroup() match {
+          case None =>
+            consumerGroupOutputWriter.printError(s"The consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' does not exist.")
+          case Some(assignments) =>
+            if (assignments.isEmpty)
+              consumerGroupOutputWriter.printError(s"Consumer group '${opts.options.valuesOf(opts.groupOpt).asScala.head}' is rebalancing.")
```

Comment:
It would be more ideal if we could tell the user which is the case. Maybe we need to propagate the state of the group down to this method.